vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSequencesController.java

```diff
@@ -54,12 +61,15 @@ public String sequence(Model model, @PathVariable("sequenceId") String sequenceI
         if (sequenceExist==null) {
             throw new SequenceNotFoundException(sequenceId);
         }
-        
+
         List<ISlide> slides = sequenceManager.getSequence(sequenceId).getSlides();
         if (slides.size() == 0) {
             throw new SlidesInSequenceNotFoundException();
         }
+        if(branchingPointId!=null){
+            choiceHistory.addToSequenceSlideHistory(previousSequenceId,branchingPointId);
+        }
```

Comment:
I would put this logic into the slide controller. otherwise you assume that the sequence controller will always be called first and then redirect but you can't guarantee that. if we change the implementation to directly link to the branching point, the history function will fail.